Tasks added needed for dokuwiki to function with nginx

sryneklar opened the pull request:
A php-xml module is now being installed. Template files are currently not being used because (see open issue). Also added a task for set permissions.

Mr-Pi committed:
Initial commit

Mr-Pi committed:
Initial commit

Mr-Pi committed:
Fix links, descriptions, add LICENSE

 * add LICENSE file for CC-BY-SA
 * use https:// instead of http://
 * use general descriptions

Mr-Pi committed:
Fix markdown syntax at README.md

 * use proper markdown instead of html

dasJ committed:
Write the name of the license instead of the link

The Ansible galaxy metadata file shouldn't contain the link to the license, but rather the license name.
I just want to know what license it is, not parse the link in my head or click it.

dasJ committed:
Remove the branch name from Galaxy metadata

main is the default and setting it again is pretty useless.

Commit:
Add initial ansible files for the dokuwiki role.

sryneklar committed:
Remove sudo: true line.

Not needed

sryneklar committed:
Change variable name to "dokuwiki_filename".

Wrong name of the variable was specified.

Commit:
Change variable name from dict to prefix based

Commit:
Merge branch 'dev' of github.com:stuvusIT/dokuwiki into dev

Incorporate changes made via web interface

Commit:
Change variable names in the template.

Commit:
Fix erroneous names of variables in the template file. Ansible now runs to completion. Functionality not actually tested

Commit:
Change first line to php comment

Commit:
Add template for protected configurations and ldap. Add plugin template and activation for ldap in plugins.php

Commit:
Begin to fill out the readme.

Commit:
Fixed typos

Commit:
Add small addition.

Commit:
Merge branch 'dev' of github.com:stuvusIT/dokuwiki into dev to fix different commit histories.

Commit:
Disable ldap auth. Fill out README.md.

Commit:
Remove Example Playbook headline.

Commit:
Pull changes requested fix. Add ldap variables. local.protected.php now has variables. Added variables to main.yml. Fix README.md

Commit:
Add remote download for the dokuwiki. Add dependencies for dokuwiki. Add more variables for protected ldap configuration.

Commit:
Remove dokuwiki_filename variable because we now fetch the stable version of dokuwiki.

Commit:
Remove commentary in default file.

Commit:
Fix name of variable.

Commit:
Fix links to repos.

Commit:
Add version variable

Commit:
Add installation of php-xml for dokuwiki.

SlothOfAnarchy commented on templates/local.protected.php.j2:
```diff
@@ -20,4 +20,4 @@ $conf['auth']['ldap']['usertree']    = {{ dokuwiki_ldap_usertree}}  #'ou=People,
 $conf['auth']['ldap']['userfilter']  = {{ dokuwiki_ldap_userfilter }} #'(&(uid=%{user})(objectClass=posixAccount))';
 $conf['auth']['ldap']['groupfilter'] = {{ dokuwiki_ldap_groupfilter}} #'(&(objectClass=posixGroup)(|(gidNumber=%{gid})(memberUID=%{user})))';
- 
+#$conf['auth']['ldap']['version']    = {{ dokuwiki_ldap_version }} #3;
```
why #3?

SlothOfAnarchy commented on defaults/main.yml:
```diff
@@ -133,9 +133,11 @@
-  dokuwiki_ldap_server: ''
-  dokuwiki_ldap_usertree: 'ou=People, dc=server, dc=tld'
-  dokuwiki_ldap_grouptree: 'ou=Group, dc=server, dc=tld'
+  dokuwiki_ldap_server: 'ldaps://ldap01.faveve.uni-stuttgart.de:636'
```
defaults should not contain our LDAP connection parameters

SlothOfAnarchy commented on tasks/main.yml:
```diff
@@ -1,9 +1,17 @@
     dest: /srv/
     remote_src: True
 
+- name: Install xml-php package used by dokuwiki.
```
add task to update package cache before this

SlothOfAnarchy commented on tasks/main.yml:
```diff
@@ -1,9 +1,17 @@
+  when: ansible_pkg_mgr == "apt"
+
+- name: Fix permissions
+  shell: chown -R www-data /srv/dokuwiki-2017-02-19b/*
```
do all these files need to get different permissions?
if it's only the template, set the permissions directly there in the task that copies it.

Scriptkiddi committed:
Added vars to set default behaviour for nginx and php
Updated some wrong values in the default settings also
changed the nginx conf to a working dokuwiki one

Scriptkiddi committed:
reworked main.yml

Mr-Pi committed:
Fix syntax and typo

Scriptkiddi committed:
install.yml

Mr-Pi committed:
Merge branch 'dev' of github.com:stuvusIT/dokuwiki into dev

Scriptkiddi committed:
Merge branch 'dev' of github.com:stuvusIT/dokuwiki into dev

Scriptkiddi committed:
renamed yaml to yml

Scriptkiddi committed:
Added step to copy the dokuwiki extract folder to the normal path
also removed the permission fix call to the shell since file and
unarchive support setting the owner

SlothOfAnarchy commented on tasks/install.yml:
```diff
@@ -0,0 +1,18 @@
+    url: https://download.dokuwiki.org/src/dokuwiki/dokuwiki-stable.tgz
+    dest: /tmp/dokuwiki-stable.tgz
+- name: unarchive dokuwiki-stable.tgz to /srv/dokuwiki
+  unarchive:
```
are old configurations and files being overridden with this task?

SlothOfAnarchy commented on tasks/install.yml:
```diff
@@ -0,0 +1,18 @@
+---
+- name: download latest dokuwiki
```
can you check if downloading is even necessary, i.e. the versions differ?

SlothOfAnarchy commented on tasks/install.yml:
```diff
@@ -0,0 +1,18 @@
+    src: /tmp/dokuwiki-stable.tgz
+    dest: /srv/
+    remote_src: yes
+
```
please add an update cache task

SlothOfAnarchy commented on vars/main.yml:
```diff
@@ -0,0 +1,44 @@
+domain_prefixes:
```
seems like a duplicate of `nginx.yml`

SlothOfAnarchy commented on vars/nginx.yml:
```diff
@@ -0,0 +1,39 @@
+      - 172.27.10.0/24
+      - 172.27.100.0/24
+    https: true
+    root: /srv/dokuwiki-2017-02-19b
```
why the date?

Scriptkiddi committed:
added pools to vars/main.yml

Mr-Pi committed:
Fix dokuwiki target

 * remove debug statement
 * create a link of current version to /srv/dokuwiki

Mr-Pi committed:
Merge branch 'dev' of github.com:stuvusIT/dokuwiki into dev
